include/rocksdb/cache.h

```diff
@@ -60,6 +73,22 @@ struct LRUCacheOptions {
   // insert to the cache will fail when cache is full.
   bool strict_capacity_limit = false;
 
+  // If non-nullptr will use this allocator instead of system allocator when
+  // allocating memory for cache blocks. Call this method before you start using
+  // the cache!
+  //
+  // Caveat: when the cache is used as block cache, the memory allocator is
+  // ignored when dealing with compression libraries that allocate memory
+  // internally (currently only XPRESS).
+  std::shared_ptr<MemoryAllocator> memory_allocator;
```

Comment:
What's the reason to move it into the CacheOptions structure? Do we pair the memory_allocator with a certain option?

Reply:
It was already in the LRUCacheOptions.